### ISSUE TITLE
Tabs groups only appear in main window

### ISSUE THREAD

**Issue by Qantas94Heavy** [projectdelphai/panorama-tab-groups#54]
Currently have three windows open. Only the first opened window shows the correct tabs groups. The rest of the windows only have one empty group even though there are tabs open. Pinned tabs still show correctly though.

Browser: Firefox Developer Edition 65.0b7 64-bit (Mozilla build)
OS: openSUSE Tumbleweed 20181224
Extension version: 0.8.6

**Comment by keeferrourke** [comment]
I would also really like to see this behaviour change. Tab management is a pain across multiple windows when this extension is enabled.

**Comment by vaxul** [comment]
How do you use multiple windows in combination with tab-groups? F.e. the session handling (closing and restoring windows/Firefox).
How do you expect the extension to work?

**Comment by keeferrourke** [comment]
@vaxul I expect to be able to see all inactive tab groups if I click the "Panorama Group Tabs" button, regardless of which window I click it from. I'd like to be able to, from another window, select a group of tabs from this extension and bring them into focus.

i.e. I can have my "main" Firefox window open on one virtual desktop, and on a secondary virtual desktop open another Firefox window. Let's say that the "main" window is handling a tab group called 'To-do', but I'd like to pull up my "Reference" group of tabs in the secondary window.

I expect to be able to, from the secondary window:
 1. Click the "Panorama Group Tabs" button in my toolbar;
 2. Be shown all "inactive" tab groups, (i.e. tab groups which are not open in another window, such as "To-do" which is open in my main window);
 3. Select my "Reference" group;
 4. See that the "Reference" tab group has been opened in my secondary window, while my "To-do" group of tabs is still open in my "main" window.

Basically without this behaviour, this extension breaks multiple-window workflows, especially across virtual desktops.

**Comment by vaxul** [comment]
@Qantas94Heavy 

I can't reproduce this behaviour. I created 3 windows with each two groups with each two tabs. I activated "restore session" in Firefox settings. If I quit Firefox (STRG+SHIFT+Q, or Menu > Quit), the reopen Firefox again, all 3 windows with each group and tabs get restored.

But your problem sound similar to #55. Is this true?

**Comment by projectdelphai** [comment]
@vaxul I think @Qantas94Heavy is asking for the same thing that @keeferrourke is asking for.

@keeferrourke / @Qantas94Heavy, I know this is a problem already, however, I haven't exactly looked into how to solve this yet. Without any research, I'm not sure how to do it since when you open a single tab group, it automatically hides all other tab groups and I'm not sure how that'll work if you have more than one tab group open (say two groups open in two windows). This is assuming that we can get multiple windows to see the same view. I'll take a further look into this when I can.

**Comment by quantitative-technologies** [comment]
I'm just trying this out. 

Do I understand correctly that if I have one Window with 2 groups: GroupA and GroupB, and then I decide I want 2 windows, say one window on each monitor, one with GroupA and the other with GroupB, this is not currently possible (of course I mean without recreating GroupA and GroupB)?

**Comment by vaxul** [comment]
Each window creates an own independent PTG stack.

**Comment by motey** [comment]
Maybe a compromise would be if we could click "Move TabGroup to another window". Then you would move the group from one PTG stack to another. Thats maybe easier/possible to realize (compared to syncing the stacks) and would help with the tab managment on multiple windows.

**Comment by orbatos** [comment]
@motey Your proposal provides exactly the needed functionality to integrate multi-window and tab group functionality simply in my view.

@projectdelphai You have the right idea about @keeferrourke 's request. This describes the potential workflow that I remember being proposed when work on Panorama was first introduced. That said, since the PTG stacks are currently separate per window (for session compatibility?), incorporating motey's workaround at least sounds simpler and would provide the necessary functionality.

**Comment by grahamperrin** [comment]
https://github.com/projectdelphai/panorama-tab-groups/issues/54#issuecomment-451726469

@projectdelphai might it help to consider how some things are done with ([MPL 2.0](https://github.com/Drive4ik/simple-tab-groups/blob/master/LICENSE)) Simple Tab Groups? 

Cherry-picked from the commit history (I chose to not go too far back): 

https://github.com/Drive4ik/simple-tab-groups/commit/67f12592b26b964835a44b1900ac870d6505a089

**Comment by grahamperrin** [comment]
https://github.com/projectdelphai/panorama-tab-groups/issues/54#issuecomment-575712823

Generally +1

> … "Move TabGroup to another window". …

The challenge here, conceptually, is how to identify each window in a way that is distinctive and user-friendly. 

For example: I often have to think twice, or three times, before selecting the correct window with **Tab Mover**. The screenshot at https://addons.mozilla.org/addon/tab-mover/ is realistic if you have only two windows … what if you have a dozen or so? I found it not _too_ difficult with that many windows, because I have a desktop environment that allows good use of title bars. Other desktop environments are less friendly. 

----

With Simple Tab Groups: 

1. open a new window (or bring to front the required window)
2. load the required group

– step (1) is free from the ambiguity that I find with, for example, Tab Mover.
